File: 06-TREES/binary-tree-inorder-traversal.py
from typing import *
from TreeNode import TreeNode

# https://leetcode.com/problems/binary-tree-inorder-traversal/
class Solution:
    def __init__(self):
        self.inordered = []

    def inorderTraversal(self, root: TreeNode | None) -> List[int]:
        if root:
            self.inorderTraversal(root.left)
            self.inordered.append(root.val)
            self.inorderTraversal(root.right)

        return self.inordered

File: 06-TREES/PHP/binary-tree-inorder-traversal.php
<?php
use TreeNode;

# https://leetcode.com/problems/binary-tree-inorder-traversal/
class Solution
{
    private array $inordered = [];

    public function inorderTraversal(?TreeNode $root): array
    {
        if ($root) {
            $this->inorderTraversal($root->left);
            $this->inordered[] = $root->val;
            $this->inorderTraversal($root->right);
        }

        return $this->inordered;
    }
}